flagengine/segments/models.go

```diff
@@ -0,0 +1,58 @@
+package segments
+
+import (
+	"github.com/Flagsmith/flagsmith-go-client/flagengine/features"
+	"github.com/Flagsmith/flagsmith-go-client/flagengine/utils"
+	"regexp"
+	"strings"
+)
+
+type SegmentConditionModel struct {
+	Operator ConditionOperator `json:"operator"`
+	Value    string            `json:"value"`
+	Property string            `json:"property_"`
+}
+
+func (m *SegmentConditionModel) MatchesTraitValue(traitValue string) bool {
+	switch m.Operator {
+	case NotContains:
+		return !strings.Contains(traitValue, m.Value)
+	case Regex:
+		return m.regex(traitValue)
+	default:
+		return match(m.Operator, traitValue, m.Value)
+	}
+}
+
+func (m *SegmentConditionModel) regex(traitValue string) bool {
+	match, err := regexp.Match(m.Value, []byte(traitValue))
+	if err != nil {
+		// TODO(tzdybal): how to handle this?
+		return false
```

Comment:
Seems fine to me to just return false here, what were you thinking otherwise?

Reply:
Propagating error to let caller know, that regex is invalid (syntactically).